kernel/arch/dreamcast/include/dc/pvr.h

```diff
@@ -17,1773 +18,206 @@
+        - PVR_MIPBIAS_0 
+            - Does it exist? What happens if set to 0?
+        - pvr_list_flush() implementation
+        - pvr_vertbuf_written()
+            - shouldn't this at least assert if you write to much? Error code?
+        - pvr_vertbuf_remaining()
+            - query for the amount of storage remaining in a vertex buffer?
 
-    \param  a               Alpha value of the fog
-    \param  r               Red value of the fog
-    \param  g               Green value of the fog
-    \param  b               Blue value of the fog
+    \author Megan Potter
+    \author Roger Cattermole
+    \author Paul Boese
+    \author Brian Paul
+    \author Lawrence Sebald
+    \author Benoit Miller
+    \author Ruslan Rostovtsev
+    \author Falco Girgis
 */
-void pvr_fog_vertex_color(float a, float r, float g, float b);
 
-/** \brief   Set the fog far depth.
-    \ingroup pvr_fog
+#ifndef __DC_PVR_H
+#define __DC_PVR_H
 
-    This function sets the PVR_FOG_DENSITY register appropriately for the
-    specified value.
+#include <stdint.h>
+#include <stdbool.h>
 
-    \param  d               The depth to set
-*/
-void pvr_fog_far_depth(float d);
+#include <sys/cdefs.h>
+__BEGIN_DECLS
 
-/** \brief   Initialize the fog table using an exp2 algorithm (like GL_EXP2).
-    \ingroup pvr_fog
+#include <stdalign.h>
 
-    This function will automatically set the PVR_FOG_DENSITY register to
-    259.999999 as a part of its processing, then set up the fog table.
+#include <arch/memory.h>
+#include <arch/types.h>
+#include <arch/cache.h>
+#include <dc/sq.h>
 
-    \param  density         Fog density value
+/** \defgroup pvr   PowerVR
+    \brief          Low-level PowerVR GPU Driver.
+    \ingroup        video
 */
-void pvr_fog_table_exp2(float density);
-
-/** \brief   Initialize the fog table using an exp algorithm (like GL_EXP).
-    \ingroup pvr_fog
 
-    This function will automatically set the PVR_FOG_DENSITY register to
-    259.999999 as a part of its processing, then set up the fog table.
-
-    \param  density         Fog density value
-*/
-void pvr_fog_table_exp(float density);
+/** PVR Primitive List Type
+    \ingroup pvr_list_mgmt
+    
+    This type is an identifier for the different kinds of primitive lists.
 
-/** \brief   Initialize the fog table using a linear algorithm (like GL_LINEAR).
-    \ingroup pvr_fog
+    \note
+    Each primitive processed by the PVR is submitted to one of the hardware
+    primitive lists. There is a separate list for each transparency type for
+    both primitives (polygons or sprites) and modifier volumes. 
+*/
+typedef enum pvr_list {
+    PVR_LIST_OP_POLY = 0,  /**< Opaque polygon list */
+    PVR_LIST_OP_MOD  = 1,  /**< Opaque modifier list */
+    PVR_LIST_TR_POLY = 2,  /**< Translucent polygon list */
+    PVR_LIST_TR_MOD  = 3,  /**< Translucent modifier list*/
+    PVR_LIST_PT_POLY = 4,  /**< Punch-thru polygon list */
+    PVR_LIST_COUNT   = 5   /**< Number of list types */
```

Comment:
No need for the explicit numbering. The same goes for every other enum.

Reply:
You're right, but... now that I looked back, I remembered why I did it... Because their values actually do matter when the PVR itself is expecting them to be equal to certain things, and the values are going directly to the HW. I was worried not making them explicit would make it look like it's safe to change the values while explicitly enumerating would make it more clear that they are fixed?

...but I'm not very strong on this opinion. I actually dislike explicit enumeration like this in general. If you don't think this is an exception that warrants it, then I'm going with your vote, because I don't feel strongly either way.

Reply:
The way I see it... those enums represent hardware values and as such shouldn't be changed, but:
- if for some reason we need new values we'd add them to the end, so it wouldn't be a problem;
- a PR that casually re-orders the enum entries would most likely cause breakages and wouldn't pass peer review (not my review at least).

Reply:
Done.